libs/wingsdk/src/target-sim/api.inflight.ts

```diff
@@ -162,7 +162,7 @@ export class Api
   }
 
   private async saveState(state: StateFileContents): Promise<void> {
-    await fs.promises.writeFile(
+    fs.writeFileSync(
```

Comment:
What's the motivation to change to sync?

(Curious)

Reply:
Here's what I believe is happening today:

1. User stops the Wing Console (via SIGTERM)
2. Wing Console asks the simulator to stop
3. Simulator starts stopping resources
4. The bucket's save() method is called.
5. The save method calls writeFile(). It creates and opens the file.
6. Node is waiting for I/O operations to finish.
7. The Wing Console process, seeing no more synchronous operations left on its call stack, exits. The writeFile() operation is unfinished so an empty (or partially written) file is saved to the state.

I haven't figured out the root cause of 7 yet (I've reached out to @skyrpex) so I opened this as a stop gap fix. Changing to sync means the file either gets written or it doesn't, which the bucket does know how to handle.